pkg/idp/provider/azure/errors.go

```diff
@@ -0,0 +1,48 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
+)
+
+var invalidConfigError = &microerror.Error{
+	Kind: "invalidConfigError",
+}
+
+// IsInvalidcConfig asserts invalidConfigError.
+func IsInvalidConfig(err error) bool {
+	return microerror.Cause(err) == invalidConfigError
+}
+
+var notFoundError = &microerror.Error{
+	Kind: "notFoundError",
+}
+
+// IsNotFound asserts notFoundError.
+func IsNotFound(err error) bool {
+	return microerror.Cause(err) == notFoundError
+}
+
+var requestFailedError = &microerror.Error{
+	Kind: "requestFailedError",
+}
+
+// IsRequestFailed asserts requestFailedError.
+func IsRequestFailed(err error) bool {
+	return microerror.Cause(err) == requestFailedError
+}
+
+func printOdataError(err error) string {
```

Comment:
Helper function to display more information when printing the microsoft graph api errors